### PR TITLE
Review of Develop

### PR DESCRIPTION
**Pull request by cppcooper**
It's time to merge into master. All this code needs review.

### CODE REVIEW

**Review by brianzhouzc — COMMENTED** [review]
We probably should merge to master more often (maybe twice a week?)

**Comment by cppcooper** [comment]
I was thinking once every two weeks, but it doesn't really matter so long as master build and works

File: include/configuration.h [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #pragma once
-#define ORTHANC_PLUGIN_NAME "DataAnonymizer"
+#define ORTHANC_PLUGIN_NAME "data-anonymizer"
 #define ORTHANC_PLUGIN_VERSION "0.1"
```
Comment (brianzhouzc) [inline include/configuration.h:3]: Update version maybe?

File: src/storage-area.cpp [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    if(file.is_open()){
+        file.seekg(rangeStart);
+        file.read((char*)target->data, target->size);
+        return OrthancPluginErrorCode_Success;
+    }
+    return OrthancPluginErrorCode_InexistentFile;
+}
+
+OrthancPluginErrorCode StorageRemoveCallback(const char *uuid, OrthancPluginContentType type) {
+    switch(type){
```
Comment (brianzhouzc) [inline src/storage-area.cpp:192]: Other file types? Maybe use GetPath directly to retrieve file path

**Review by brianzhouzc — CHANGES_REQUESTED** [review]
Looks good to me. I would increment the version number. Other issues are being worked on in other branches, I would open some issues for all the todos.

**Review by brianzhouzc — DISMISSED** [review]

File: src/storage-area.cpp (hunk 2 above)
Comment (cppcooper) [reply src/storage-area.cpp:192]: GetPath doesn't really work for the other types

File: include/configuration.h (hunk 1 above)
Comment (cppcooper) [reply include/configuration.h:3]: This will be in PR #87

File: tests/tests-libraries.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    //    and paramspec is:
+    //     name=value[&...]
+    // ex:
+    //  postgresql://user:secret@localhost
+    //  postgresql://other@localhost/otherdb?connect_timeout=10&application_name=myapp
+    //  postgresql://host1:123,host2:456/somedb?target_session_attrs=any&application_name=myapp
+    //pqxx::connection c("postgresql://postgres:example@localhost:5432");
+    //pqxx::work w(c);
+    DBInterface::connect("example");
+    ASSERT_TRUE(DBInterface::is_open());
```
Comment (iwanlevin) [inline tests/tests-libraries.cpp:67]: We should probably close this connection before exiting the test

Comment (cppcooper) [reply tests/tests-libraries.cpp:67]: I'll implement this on 87 as to avoid a new pr

Comment (iwanlevin) [reply tests/tests-libraries.cpp:67]: sounds good

**Review by iwanlevin — DISMISSED** [review]

**Review by raouft — DISMISSED** [review]

**Review by brianzhouzc — APPROVED** [review]

**Review by raouft — APPROVED** [review]

**Review by iwanlevin — APPROVED** [review]

**Pull request merged**